crates/astria-composer/src/searcher/mod.rs

```diff
@@ -44,8 +50,10 @@ pub(super) struct Searcher {
     collectors: HashMap<String, Collector>,
     collector_statuses: HashMap<String, watch::Receiver<collector::Status>>,
     // A channel on which the searcher receives transactions from its collectors.
-    new_transactions: Receiver<collector::Transaction>,
-    collector_tasks: tokio_util::task::JoinMap<String, eyre::Result<()>>,
+    new_transactions_rx: Receiver<collector::Transaction>,
+    new_transactions_tx: Sender<collector::Transaction>,
+    rollups: HashMap<String, String>,
+    collector_tasks: JoinMap<String, eyre::Result<()>>,
```

Comment:
same here (i guess chain_id to result?)

Reply:
Added a comment. It's actually just a mechanism to track if a collector task is still running. `collector_tasks.join_next()` should only return `Some(exit_status)` if the collector exited for a reason like the websocket going down.